crates/runtime/src/memory.rs

```diff
@@ -5,17 +5,18 @@
 use crate::mmap::Mmap;
 use crate::vmcontext::VMMemoryDefinition;
 use more_asserts::{assert_ge, assert_le};
+use std::cell::{Cell, RefCell};
 use std::convert::TryFrom;
 use wasmtime_environ::{MemoryPlan, MemoryStyle, WASM_MAX_PAGES, WASM_PAGE_SIZE};
 
 /// A linear memory instance.
 #[derive(Debug)]
 pub struct LinearMemory {
     // The underlying allocation.
-    mmap: Mmap,
+    mmap: RefCell<Mmap>,
 
     // The current logical size in wasm pages of this linear memory.
-    current: u32,
+    current: Cell<u32>,
```

Comment:
Logically, `mmap` and `current` here should both be part of a single cell, since they need to be kept in sync with each other, so that someone doesn't mutate `current`` while someone else has an `&mut` for `mmap`. Would it make sense to factor out these two fields into a separate struct so that they could share a `RefCell`?

Reply:
Sounds like a good idea to me!

Reply:
Ok I've done this for now, but as a general idioms thing I think it's fine either way. Here the abstraction boundary is moreso `LinearMemory` which doesn't expose its interior-ly mutable fields, but rather all of its methods obey certain invariants.

I think that eventually we'll want to go back to individually mutable fields for when we add threads support, for example. There fetching the current size should probably just be an atomic load rather than acquiring a mutex (as would be the equivalent of `RefCell`).

In any case just wanted to mention that I think there's rationale for both, but for now I think wrapping it all up in one struct makes more sense.